### PR TITLE
Feature 145 support for facilities

### LINKED ISSUE #145: Add support for facilities
As a user I'd like to have option to select, which facility's data I'm looking for.

### PR DESCRIPTION
**Pull request by EVP1989**
Resolves #145

### CODE REVIEW

File: src/components/EditSeedBatch.tsx [hunk 1]
```diff
@@ -81,20 +82,24 @@ class EditSeedBatch extends React.Component<Props, State> {
...
         return;
       }      
-      const seedBatchesService = await Api.getSeedBatchesService(this.props.keycloak);
-      const seedsService = await Api.getSeedsService(this.props.keycloak);      
-      const seedBatch = await seedBatchesService.findSeedBatch({seedBatchId: this.props.seedBatchId});
-      this.props.onSeedBatchSelected && this.props.onSeedBatchSelected(seedBatch);
+      const seedBatchesService = await Api.getSeedBatchesService(keycloak);
+      const seedsService = await Api.getSeedsService(keycloak);      
+      const seedBatch = await seedBatchesService.findSeedBatch({
+        seedBatchId: this.props.seedBatchId,
```
Comment (Jdallos) [inline src/components/EditSeedBatch.tsx:93]: Can you not destructure seedBatchId from props on line 85 as well?

File: src/components/FacilitySelect.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+class FacilitySelect extends React.Component<Props, State> {
+
+  /**
+   * Constructor
+   * @param props component properties 
+   */
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+    }
```
Comment (Jdallos) [inline src/components/FacilitySelect.tsx:36]:
```suggestion
    this.state = {}
```

File: src/components/FacilitySelect.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  }
+
+  /**
+   * Render facility select
+   */
+  render() {
+    const { facility, keycloak, onFacilityChange } = this.props;
+    if (!facility || !keycloak?.hasRealmRole("admin")) {
```
Comment (Jdallos) [inline src/components/FacilitySelect.tsx:44]: Are we using keycloak role of admin?

File: src/components/FacilitySelect.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+    if (keycloak.hasRealmRole("juva") && keycloak.hasRealmRole("joroinen")) {
+      return (
+        <Dropdown position="right" item text={ facility }>
+          <Dropdown.Menu>
+            <Dropdown.Item
+              value={ facility === Facility.Joroinen ? Facility.Juva : Facility.Joroinen }
+              onClick={ (event, item) => { onFacilityChange(item.value as Facility) } }
+            >
+              { facility === Facility.Joroinen ? Facility.Juva : Facility.Joroinen }
```
Comment (Jdallos) [inline src/components/FacilitySelect.tsx:56]: Perhaps I'm misunderstanding this, if the facility is Joroinen wouldnt we want to display the facility as Joroinen?

File: src/components/PackageSizeList.tsx [hunk 5]
```diff
@@ -19,8 +19,9 @@ import LocalizedUtils from "../localization/localizedutils";
 export interface Props {
   keycloak?: Keycloak.KeycloakInstance;
   packageSizes?: PackageSize[];
+  facility: Facility;
   onPackageSizesFound?: (packageSizes: PackageSize[]) => void,
-   onError: (error: ErrorMessage | undefined) => void
+  onError: (error: ErrorMessage | undefined) => void
```
Comment (Jdallos) [inline src/components/PackageSizeList.tsx:24]:
```suggestion
  onError: (error: ErrorMessage | undefined) => void;
```

File: src/components/PerformedCultivationActionList.tsx [hunk 6]
```diff
@@ -20,8 +20,9 @@ import LocalizedUtils from "../localization/localizedutils";
 export interface Props {
   keycloak?: Keycloak.KeycloakInstance;
   performedCultivationActions?: PerformedCultivationAction[];
+  facility: Facility;
   onPerformedCultivationActionsFound?: (performedCultivationActions: PerformedCultivationAction[]) => void,
-   onError: (error: ErrorMessage | undefined) => void
+  onError: (error: ErrorMessage | undefined) => void
```
Comment (Jdallos) [inline src/components/PerformedCultivationActionList.tsx:25]:
```suggestion
  onError: (error: ErrorMessage | undefined) => void;
```

File: src/components/WelcomePage.tsx [hunk 7]
```diff
@@ -76,17 +79,24 @@ class WelcomePage extends React.Component<Props, any> {
...
     const kcConf = {
       "realm": process.env.REACT_APP_KEYCLOAK_REALM as string,
       "url": process.env.REACT_APP_AUTH_SERVER_URL as string,
       "clientId": process.env.REACT_APP_AUTH_RESOURCE as string
     };
     const keycloak = new Keycloak(kcConf);
-    keycloak.init({onLoad: "login-required", checkLoginIframe: false}).success((authenticated) => {
-      this.props.onLogin && this.props.onLogin(keycloak, authenticated);
+    await keycloak.init({onLoad: "login-required", checkLoginIframe: false}).success((authenticated) => {
+    onLogin && onLogin(keycloak, authenticated);
```
Comment (Jdallos) [inline src/components/WelcomePage.tsx:91]:
```suggestion
      onLogin && onLogin(keycloak, authenticated);
```

File: src/components/WelcomePage.tsx [hunk 8]
```diff
@@ -76,17 +79,24 @@ class WelcomePage extends React.Component<Props, any> {
...
+    const { onFacilityUpdate, onLogin } = this.props;
     const kcConf = {
       "realm": process.env.REACT_APP_KEYCLOAK_REALM as string,
       "url": process.env.REACT_APP_AUTH_SERVER_URL as string,
       "clientId": process.env.REACT_APP_AUTH_RESOURCE as string
     };
     const keycloak = new Keycloak(kcConf);
-    keycloak.init({onLoad: "login-required", checkLoginIframe: false}).success((authenticated) => {
-      this.props.onLogin && this.props.onLogin(keycloak, authenticated);
+    await keycloak.init({onLoad: "login-required", checkLoginIframe: false}).success((authenticated) => {
```
Comment (Jdallos) [inline src/components/WelcomePage.tsx:90]: Interesting that this was working before without being async and awaited.

File: src/generated/client/.openapi-generator/FILES [hunk 9]
```diff
@@ -1,3 +1,4 @@
+.openapi-generator-ignore
```
Comment (Jdallos) [inline src/generated/client/.openapi-generator/FILES:1]: As discussed previously, I believe all generated files can be removed from version control/ added to the .gitignore, This is what was requested for the app

**Review by Jdallos — CHANGES_REQUESTED** [review]
Some minor comments and suggestions

File: src/components/WelcomePage.tsx (hunk 8 above)
Comment (villejuutila) [reply src/components/WelcomePage.tsx:90]: @Jdallos .success() is a legacy method for handling promises, similar to .then() and .catch() etc.

File: src/generated/client/.openapi-generator/FILES (hunk 9 above)
Comment (EVP1989) [reply src/generated/client/.openapi-generator/FILES]: Done

File: src/components/WelcomePage.tsx (hunk 8 above)
Comment (Jdallos) [reply src/components/WelcomePage.tsx:90]: Ok, so .success() is deprecated, could we change this functionality to remove this?

**Review by Jdallos — APPROVED** [review]

File: src/components/WelcomePage.tsx (hunk 8 above)
Comment (EVP1989) [reply src/components/WelcomePage.tsx:90]: It can be done in the future but I think in this issue, it would be out of scope.

**Pull request merged**